### PR TITLE
Rename package to ansible-base

### PR DESCRIPTION
**Pull request by abadger**
##### SUMMARY
Rename the package from ansible to ansible-base

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

Ansible build scripts:
* setup.py

##### ADDITIONAL INFORMATION
In order for people to test the ACD package that I'm shooting to have ready by the end of the week, we'll need  to have an ansible-base package that people can install.  We'll need the package names to not conflict in order to install from git.  Would be even nicer if we could upload a version of this to testpypi (maybe changing the version so that it doesn't conflict with our eventual 2.10.0 alphas?)

@webknjaz Could you look this over and make sure I didn't miss anything?  I think this is all that has to be done to make setup.py do the right thing but I am a little leary since I had done some things wrong with the `package_dir` key before.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -292,7 +292,7 @@ def get_dynamic_setup_params():
     # Ansible will also make use of a system copy of python-six and
     # python-selectors2 if installed but use a Bundled copy if it's not.
     python_requires='>=2.7,!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*,!=3.4.*',
-    package_dir={'': 'lib',
+    package_dir={'ansible': 'lib/ansible',
```
Comment (webknjaz) [inline setup.py]: According to https://setuptools.readthedocs.io/en/latest/setuptools.html#using-a-src-layout, the old version should be enough... Does it cause problems for you?
```suggestion
    package_dir={'': 'lib',
```

File: setup.py [hunk 2]
```diff
@@ -274,7 +274,7 @@ def get_dynamic_setup_params():
         'install_scripts': InstallScriptsCommand,
         'sdist': SDistCommand,
     },
-    name='ansible',
+    name='ansible-base',
```
Comment (webknjaz) [inline setup.py:260]: FTR this specific part looks good to me. The confusion comes with `package_dir` and understanding the full picture and the mention of Git.

**Comment by webknjaz** [comment]
> we could upload a version of this to testpypi (maybe changing the version so that it doesn't conflict with our eventual 2.10.0 alphas?)

I normally set it up so that the versions are dev-versions with a number that's a distance to the most recent tag in the branch. Example: https://test.pypi.org/p/ansible-lint#history.

**Comment by webknjaz** [comment]
> We'll need the package names to not conflict in order to install from git

People can `pip install git+https://github.com/ansible/ansible.git#egg=ansible-base` and the name in `pip list` will be `ansible-base` if that's what you're looking for.

File: setup.py (hunk 1 above)
Comment (abadger) [reply setup.py]:
```
[pts/33@peru /srv/ansible/abadger8]$ python3 setup.py sdist                                                                                                                                                    *[rename-tarball]  (10:17:22)
running sdist
running check
error: package directory 'lib/ansible/ansible' does not exist
```

So yeah, that's why I had to change that line.

**Comment by abadger** [comment]
Thanks! That should work for testing. This does need to get merged before 2.10 starts making releases, though, so the earlier we can get the exact methods worked out and get it merged, the better.

File: setup.py (hunk 1 above)
Comment (webknjaz) [reply setup.py]: Hm.. This looks wrong

Comment (webknjaz) [reply setup.py]: @abadger I just checked locally and cannot reproduce your error. Besides, it's recommended to run `twine check` and not `setup.py check`.

Comment (abadger) [reply setup.py]: I don't know what to tell you....  You started with this PR and then you changed package_dir back to `{'': 'lib',`  ?

Comment (webknjaz) [reply setup.py]: yep

Comment (abadger) [reply setup.py]: webknjaz gave me a video and I figured out what I was testing wrong... I changed the key back but not the value.   Pushed a fix now.

**Review by mattclay — APPROVED** [review]

**Review by webknjaz — APPROVED** [review]

**Comment by webknjaz** [comment]
Just to be clear. The command below should work for testing even w/o this patch:
> People can `pip install git+https://github.com/ansible/ansible.git#egg=ansible-base` and the name in `pip list` will be `ansible-base` if that's what you're looking for.

**Comment by abadger** [comment]
Merged.

**Pull request merged**